pkg/util/kube.go

```diff
@@ -419,6 +419,38 @@ func GetPodName(n, labelSelector string) (pod string, err error) {
 	return
 }
 
+// Check PodReady (Example: 2/2 container running on a pod) returns true if the pod is ready. Params: namespace, pod label and timeout to check ready pod.
+func CheckPodReady(n, labelSelector string, timeout int) (ready bool, err error) {
+	ready = false
+	for i := 0; i < timeout; i++ {
```

Comment:
This logic looks correct in general, but I suggest you to check how we approach this kind of problems in the Istio code base: https://github.com/maistra/istio/blob/maistra-2.4/tests/integration/servicemesh/smmr_test.go#L81.
If you had function like `retry.UntilSuccessOrFail`, you could make readiness checks more flexible.

Reply:
We will use retry.Retry that is implemented now